packages/video_player/video_player/lib/src/web_vtt.dart

```diff
@@ -0,0 +1,168 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'closed_caption_file.dart';
+import 'package:html/parser.dart';
+
+/// Represents a [ClosedCaptionFile], parsed from the WebVTT file format.
+/// See: https://en.wikipedia.org/wiki/WebVTT
+class WebVTTCaptionFile extends ClosedCaptionFile {
+  /// Parses a string into a [ClosedCaptionFile], assuming [fileContents] is in
+  /// the WebVTT file format.
+  /// * See: https://en.wikipedia.org/wiki/WebVTT
+  WebVTTCaptionFile(this.fileContents)
+      : _captions = _parseCaptionsFromWebVTTString(fileContents);
+
+  /// The entire body of the VTT file.
+  final String fileContents;
+
+  @override
+  List<Caption> get captions => _captions;
+
+  final List<Caption> _captions;
+}
+
+List<Caption> _parseCaptionsFromWebVTTString(String file) {
+  final List<Caption> captions = <Caption>[];
+  int number = 1;
+  for (List<String> captionLines in _readWebVTTFile(file)) {
+    if (captionLines.length < 2) continue;
+    print(captionLines);
+
+    final int captionNumber = number;
+    final _StartAndEnd startAndEnd =
+        _StartAndEnd.fromWebVTTString(captionLines[0]);
+
+    final String text = captionLines.sublist(1).join('\n');
+
+    //TODO: Handle text format
+    final String textWithoutFormat = _parseHtmlString(text);
+
+    final Caption newCaption = Caption(
+      number: captionNumber,
+      start: startAndEnd.start,
+      end: startAndEnd.end,
+      text: textWithoutFormat,
+    );
+
+    if (newCaption.start != null && newCaption.end != null) {
+      captions.add(newCaption);
+      number++;
+    }
+  }
+
+  return captions;
+}
+
+class _StartAndEnd {
+  final Duration start;
+  final Duration end;
+
+  _StartAndEnd(this.start, this.end);
+
+  // Assumes format from an VTT file.
+  // For example:
+  // 00:09.000 --> 00:11.000
+  static _StartAndEnd fromWebVTTString(String line) {
+    final RegExp format =
+        RegExp(_webVTTTimeStamp + _webVTTArrow + _webVTTTimeStamp);
+
+    if (!format.hasMatch(line)) {
+      return _StartAndEnd(null, null);
+    }
+
+    final List<String> times = line.split(_webVTTArrow);
+
+    final Duration start = _parseWebVTTTimestamp(times[0]);
+    final Duration end = _parseWebVTTTimestamp(times[1]);
+
+    return _StartAndEnd(start, end);
+  }
+}
+
+String _parseHtmlString(String htmlString) {
+  var document = parse(htmlString);
```

Comment:
nit: Flutter style guide (which the plugins mostly follow) prefer explicitly typing:

```suggestion
  final Document document = parse(htmlString);
```